WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleItemListErrorView.swift

```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+
+struct PointOfSaleItemListErrorView: View {
+    private var viewModel: any ItemListViewModelProtocol
+
+    init(viewModel: any ItemListViewModelProtocol) {
+        self.viewModel = viewModel
+    }
+
+    var errorContents: ItemListViewModel.ErrorModel {
+        guard let errorContents = viewModel.state.hasError else {
+            return ItemListViewModel.ErrorModel(title: "Unknown error",
+                                                subtitle: "Unknown error",
+                                                buttonText: "Retry")
+        }
+        return errorContents
```

Comment:
Depending on the view model for another view seems like a bit of a code smell. I know they're related, but you could potentially simplify this view by having it depend on `errorContents` and a reload action closure.

Reply:
That makes sense. I've updated this on 0655f09a1b64fe467326062ce16afed3e09f1c39 by passing the error model only and a closure. I've also moved the "unknown error" to the creation of the ErrorModel. I don't see a case where this could happen ("_there was an error retrieving the error"_), so we could just keep retuning `nil` and remove this default.